File: Source/ApplicationModel/Frontend/commands/CommandValidator.ts
// Copyright (c) Aksio Insurtech. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { Validator } from '../validation/Validator';

export type CommandPropertyValidators = { [key: string]: Validator; };

/**
 * Represents the command validator
 */
export abstract class CommandValidator {
    abstract readonly properties: CommandPropertyValidators;

    get isValid() {
        return true;
    }
}

File: Source/Infrastructure/Commands/CommandValidator.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using Cratis.Validation;

namespace Cratis.Commands;

/// <summary>
/// Represents the base type for a validator of commands.
/// </summary>
/// <typeparam name="T">Type of command.</typeparam>
public class CommandValidator<T> : DiscoverableValidator<T>;
